drv/lpc55-swd/src/main.rs

```diff
@@ -651,28 +639,32 @@ impl NotificationHandler for ServerImpl {
         //     and don't complicate the behavior here.
         //
 
-        let mut invalidate = false;
         let gpio = Pins::from(self.gpio);
 
         if (bits & notifications::JTAG_DETECT_IRQ_MASK) != 0 {
             ringbuf_entry!(Trace::SpJtagDetectFired);
             const SLOT: PintSlot = SP_TO_ROT_JTAG_DETECT_L_PINT_SLOT;
-            if let Ok(Some(detected)) =
-                gpio.pint_op(SLOT, PintOp::Detected, PintCondition::Falling)
-            {
+            if let Ok(detected) = gpio.pint_detect(SLOT, PintFlag::Falling) {
                 if detected {
                     ringbuf_entry!(Trace::InvalidateSpMeasurement);
                     // Reset the attestation log
-                    invalidate = true;
-                    self.next_use_must_setup_swd();
+                    self.invalidate_sp_measurement();
+
+                    // Cancel ongoing transactions
+                    self.transaction = None;
+                    self.state = SwdState::Disconnected;
+                    switch_io_disconnected();
                 }
+                let _ = gpio.pint_clear(SLOT, PintFlag::Both);
```

Comment:
Hmmm. It looks like we're using pin change interrupts for the falling edge, and polling detection (below) for the high state. So why is this clearing `Both`?